miner/src/pool/scoring.rs

```diff
@@ -123,16 +123,16 @@ impl<P> txpool::Scoring<P> for NonceAndGasPrice where P: ScoredTransaction + txp
 	}
 
 	fn should_replace(&self, old: &P, new: &P) -> scoring::Choice {
-		if old.sender() == new.sender() {
+		if old.priority().is_local() && new.priority().is_local() {
+			// accept local transactions over the limit
+			scoring::Choice::InsertNew
```

Comment:
Sorry for confusion, but now that I think about it we should probably handle the case when senders AND nonces are equal. In such case we need to replace transaction instead of accepting both of them

Reply:
Would be worth checking if we hit that particular branch with `should_replace` anyway, cause I think it might be handled separately by calling `choose` directly.